lib/layout/forceDirectedLayout.js

```diff
@@ -173,6 +172,36 @@ var springForce = function(edge) {
   }
 };
 
+var portLayoutHelper = function(ports, cardinalDirection, maxPortsPerSide) {
+  for(var j = 0; j < ports.length; j++) {
+    var posPercentage = 12 + 15 * (j % maxPortsPerSide);
```

Comment:
I don't think it's necesary to make the change, according to Mayur they shouldn't have more than 3 - 4 inputs/outputs on a node.